src/js/main.js

```diff
@@ -707,95 +686,89 @@ function updateTabList(features) {
 }
 
 function zeroPad(value, width) {
-    value = "" + value;
 
-    while (value.length < width) {
-        value = "0" + value;
+    let valuePadded = String(value);
+
+    while (valuePadded.length < width) {
+        valuePadded = `0${value}`;
     }
 
-    return value;
+    return valuePadded;
 }
 
 function generateFilename(prefix, suffix) {
-    var date = new Date();
-    var filename = prefix;
+    const date = new Date();
+    let filename = prefix;
 
     if (CONFIG) {
         if (CONFIG.flightControllerIdentifier) {
-            filename = CONFIG.flightControllerIdentifier + '_' + filename;
+            filename = `${CONFIG.flightControllerIdentifier}_${filename}`;
         }
         if(CONFIG.name && CONFIG.name.trim() !== '') {
-            filename = filename + '_' + CONFIG.name.trim().replace(' ', '_');
+            filename = `${filename}_${CONFIG.name.trim().replace(' ', '_')}`;
         }
     }
 
-    filename = filename + '_' + date.getFullYear()
-        + zeroPad(date.getMonth() + 1, 2)
-        + zeroPad(date.getDate(), 2)
-        + '_' + zeroPad(date.getHours(), 2)
-        + zeroPad(date.getMinutes(), 2)
-        + zeroPad(date.getSeconds(), 2);
+    const yyyymmdd = `${date.getFullYear()}${zeroPad(date.getMonth() + 1, 2)}${zeroPad(date.getDate(), 2)}`;
+    const hhmmss = `${zeroPad(date.getHours(), 2)}${zeroPad(date.getMinutes(), 2)}${zeroPad(date.getSeconds(), 2)}`;
+    filename = `${filename}_${yyyymmdd}_${hhmmss}`;
 
-    return filename + '.' + suffix;
+    return `${filename}.${suffix}`;
 }
 
 function getTargetVersion(hardwareId) {
-    var versionText = '';
+    let versionText = '';
 
     if (hardwareId) {
-       versionText += i18n.getMessage('versionLabelTarget') + ': ' + hardwareId;
+       versionText = `${i18n.getMessage('versionLabelTarget')}: ${hardwareId}`;
     }
 
     return versionText;
 }
 
 function getFirmwareVersion(firmwareVersion, firmwareId) {
-    var versionText = '';
+    let versionText = '';
 
     if (firmwareVersion) {
-        versionText += i18n.getMessage('versionLabelFirmware') + ': ' + firmwareId + ' ' + firmwareVersion;
+        versionText = `${i18n.getMessage('versionLabelFirmware')}: ${firmwareId} ${firmwareVersion}`;
     }
 
     return versionText;
 }
 
 function getConfiguratorVersion() {
-    return i18n.getMessage('versionLabelConfigurator') + ': ' + CONFIGURATOR.version;
+    return `${i18n.getMessage('versionLabelConfigurator')}: ${CONFIGURATOR.version}`;
 }
 
 function updateTopBarVersion(firmwareVersion, firmwareId, hardwareId) {
-    var versionText = getConfiguratorVersion() + '<br />';
-
-    versionText = versionText + getFirmwareVersion(firmwareVersion, firmwareId) + '<br />';
-
-    versionText = versionText + getTargetVersion(hardwareId);
+    const versionText = `${getConfiguratorVersion()}<br />${getFirmwareVersion(firmwareVersion, firmwareId)}<br />${getTargetVersion(hardwareId)}`;
```

Comment:
I can see your point about readability now. I don't think using `concat()` makes it much better though. How about

```
const configuratorVersion = getConfiguratorVersion();
const firmwareVersion = getFirmwareVersion(firmwareVersion, firmwareId);
const targetVersion = getTargetVersion(hardwareId);
const versionText = `${configuratorVersion}<br />${firmwareVersion}<br />${targetVersion}`;
```

It's a bit more wordy, but at least you can see at one glance how the finished string will look like.

Reply:
True. I think readability must be concern number one here. I will fix it later.